### ISSUE TITLE
Equational constraint produces syntactically invalid result

### ISSUE THREAD

**Issue by mx-psi** [lspitzner/brittany#267]
Running brittany 0.12.1.0 with the following program

```haskell
{-# LANGUAGE TypeFamilies #-}

f :: ((~) a b) => a -> b
f = id
```

outputs `ERROR: brittany pretty printer returned syntactically invalid result.`

**Comment by tfausak** [comment]
Here's what Brittany outputs with `--output-on-errors`:

``` hs
{-# LANGUAGE TypeFamilies #-}
f :: ((Data.Type.Equality~) a b) => a -> b
f = id
```

Also worth noting that the following equivalent program works fine:

``` hs
{-# language TypeFamilies #-}
f :: a ~ b => a -> b
f = id
```

Might be somewhat related to #258.

**Comment by lspitzner** [comment]
We already have some special-casing around this at

https://github.com/lspitzner/brittany/blob/059bb9402e207c1be0a2f7a9f0673238240531fe/src/Language/Haskell/Brittany/Internal/LayouterBasics.hs#L235-L239

I wonder why this does not trigger here.

**Comment by lspitzner** [comment]
Should be fixed in 77d6d5b553720102f05b3977f89566f2c5c1960b, will merge into master if CI is green in a few minutes.

**Comment by lspitzner** [comment]
fixed in 0.12.1.1
